### ISSUE TITLE
[arrow-parens] False positive on untyped single optional parameter

### ISSUE THREAD

**Issue by taejs** [typescript-eslint/typescript-eslint#2677]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
       "arrow-parens": ["error", "as-needed", { "requireForBlockBody": false }],
  }
}
```

```TS
const foo = (t?) => {};
```

'?' character means that it's optional parameter.
So It must not be removed executing eslint fix.

**Expected Result**
```TS
const foo = (t?) => {};
```

**Actual Result**
```TS
const foo = t? => {};
```
error is occured.
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.4.1` |
| `@typescript-eslint/parser`        | `4.4.1` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `7.11.0` |
| `node`                             | `12.16.3` |

**Comment by bradzacher** [comment]
Why are you not providing actual types for your parameters?

**Comment by taejs** [comment]
@bradzacher  That's a good point.. 
But I think we need to focus on that expression is not error. 

Also I made mistakes(I changed "Actual Result"), actual result is occurring typescript syntax error.
Is it able to be fixed?

**Comment by yeonjuan** [comment]
Hi:) @taejs 
> Repro
> {
  "rules": {
    'arrow-parens': ['error', 'always'],
  }
}

It's not occurred with the **always** option. The option in the issue should be **as-needed** for repro - [demo](https://yeonjuan.github.io/typescript-eslint-demo/#eyJjb2RlIjoiY29uc3QgZm9vID0gKHQ/KSA9PiB7fTtcbmNvbnN0IGZvbyA9ICh0PzogRm9vKSA9PiB7fTsiLCJydWxlcyI6eyJAdHlwZXNjcmlwdC1lc2xpbnQvcmV0dXJuLWF3YWl0IjoiZXJyb3IifX0=)

**Comment by taejs** [comment]
That's right. I changed the option. Thanks! @yeonjuan

**Comment by bradzacher** [comment]
also for your example code, the correct setting is `requireForBlockBody: false`.

**Comment by bradzacher** [comment]
This should really be fixed in the parent rule, as it already has some support for TS semantics.
https://github.com/eslint/eslint/blob/master/lib/rules/arrow-parens.js#L153-L154

Though I definitely would say that you should never have an untyped parameter, and should be using the `noImplicitAny` compiler option.
Using typescript without types is such a waste!

**Comment by JoshuaKGoldberg** [comment]
Closing this issue as it's been marked `external` for 2.5 years. Please post back if you think this is in error!
